File: festival-web/src/main/java/cn/bdqfork/web/annotation/RouteController.java
package cn.bdqfork.web.annotation;

import cn.bdqfork.aop.annotation.Optimize;

import javax.inject.Named;
import java.lang.annotation.*;

/**
 * @author bdq
 * @since 2020/1/29
 */
@Optimize
@Named
@RouteMapping
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE})
public @interface RouteController {
    String value() default "";
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/route/annotation/RouteController.kt
package cn.bdqfork.kotlin.web.route.annotation

import cn.bdqfork.aop.annotation.Optimize
import javax.inject.Named

/**
 * @author bdq
 * @since 2020/1/29
 */
@Optimize
@Named
@RouteMapping
@MustBeDocumented
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
annotation class RouteController(val value: String = "")